### PR TITLE
Fix setting host_tx_ready from initializePort()

### PR DESCRIPTION
**Pull request by noaOrMlnx**
**What I did**
During port initialization, we now set Host Tx Ready key-val to DB using "hset()" instead of using the new function "setHostTxReady()"

**Why I did it**
Port must be initialized before calling setHostTxReady(). if it is not initialized, it won't find the port object and function won't set host_tx_ready. 

**How I verified it**
boot switch with new changes.

**Details if related**

### CODE REVIEW

File: orchagent/portsorch.cpp [hunk 1]
```diff
@@ -5464,7 +5464,7 @@ bool PortsOrch::initializePort(Port &port)
         string hostTxReadyStr = hostTxReadyVal ? "true" : "false";
 
         SWSS_LOG_DEBUG("Received host_tx_ready current status: port_id: 0x%" PRIx64 " status: %s", port.m_port_id, hostTxReadyStr.c_str());
-        setHostTxReady(port.m_port_id, hostTxReadyStr);
+	m_portStateTable.hset(port.m_alias, "host_tx_ready", hostTxReadyStr);
```
Comment (stepanblyschak) [inline orchagent/portsorch.cpp:5467]: Please correct indent, should be using spaces

Comment (noaOrMlnx) [reply orchagent/portsorch.cpp]: Done

**Review by stepanblyschak — APPROVED** [review]

**Comment by dprital** [comment]
@prsunny @prgeor , Can you please review ?

**Comment by prsunny** [comment]
I feel there is a gap here. Why was it missed in VS test? How did we find this issue?

**Comment by noaOrMlnx** [comment]
@prsunny 
We found out about this issue while seeing the following log error - 
ERR swss#orchagent: :- setHostTxReady: Failed to get port object for port id 0x1000000000015

the issue here is that setHostTxReady requires the port to be initialize and that is why it can’t be called from initialization.

**Comment by prsunny** [comment]
@prgeor , please approve

**Comment by noaOrMlnx** [comment]
Hi @prgeor, a kind reminder

**Comment by prgeor** [comment]
> @prsunny We found out about this issue while seeing the following log error - ERR swss#orchagent: :- setHostTxReady: Failed to get port object for port id 0x1000000000015
> 
> the issue here is that setHostTxReady requires the port to be initialize and that is why it can’t be called from initialization.

@noaOrMlnx How Host Tx signal be ready without port fully initialized?

File: orchagent/portsorch.cpp [hunk 2]
```diff
@@ -5465,7 +5465,7 @@ bool PortsOrch::initializePort(Port &port)
         string hostTxReadyStr = hostTxReadyVal ? "true" : "false";
 
         SWSS_LOG_DEBUG("Received host_tx_ready current status: port_id: 0x%" PRIx64 " status: %s", port.m_port_id, hostTxReadyStr.c_str());
-        setHostTxReady(port.m_port_id, hostTxReadyStr);
+        m_portStateTable.hset(port.m_alias, "host_tx_ready", hostTxReadyStr);
```
Comment (prgeor) [inline orchagent/portsorch.cpp:5468]: @noaOrMlnx please check above in this functin, port.m_port_id is being used ...so all those are also failing?

Comment (noaOrMlnx) [reply orchagent/portsorch.cpp:5468]: @prgeor 
The issue here is not port.X, but the fact that inside setHostTxReady() function, we call getPort(). 
getPort cannot return the port object until port is initialize.

Comment (prgeor) [reply orchagent/portsorch.cpp:5468]: @noaOrMlnx Can you please remove the getPort() from setHostTxReady()? This way all host_tx_ready change will be contained in one single function

Comment (prgeor) [reply orchagent/portsorch.cpp:5468]: There is no need for portid inside setHostTxReady,

`PortsOrch::setHostTxReady(swss::Port, const std::string &status)` and modify the syslog to show only Alias (no need for port ID)

**Comment by noaOrMlnx** [comment]
Hi @prgeor @prsunny 
I  opened a new PR that does what you suggested. please check https://github.com/sonic-net/sonic-swss/pull/3133

closing the current PR.
Thanks

**Pull request closed without merging**